pkg/apiserver/experiment/experiment.go

```diff
@@ -224,49 +227,28 @@ func (s *Service) createExperiment(c *gin.Context) {
 		return
 	}
 
-	switch exp.Target.Kind {
-	case v1alpha1.KindPodChaos:
-		if err := s.createPodChaos(exp); err != nil {
-			c.Status(http.StatusInternalServerError)
-			_ = c.Error(utils.ErrInternalServer.WrapWithNoMessage(err))
-			return
-		}
-	case v1alpha1.KindNetworkChaos:
-		if err := s.createNetworkChaos(exp); err != nil {
-			c.Status(http.StatusInternalServerError)
-			_ = c.Error(utils.ErrInternalServer.WrapWithNoMessage(err))
-			return
-		}
-	case v1alpha1.KindIOChaos:
-		if err := s.createIOChaos(exp); err != nil {
-			c.Status(http.StatusInternalServerError)
-			_ = c.Error(utils.ErrInternalServer.WrapWithNoMessage(err))
-			return
-		}
-	case v1alpha1.KindTimeChaos:
-		if err := s.createTimeChaos(exp); err != nil {
-			c.Status(http.StatusInternalServerError)
-			_ = c.Error(utils.ErrInternalServer.WrapWithNoMessage(err))
-			return
-		}
-	case v1alpha1.KindKernelChaos:
-		if err := s.createKernelChaos(exp); err != nil {
-			c.Status(http.StatusInternalServerError)
-			_ = c.Error(utils.ErrInternalServer.WrapWithNoMessage(err))
-			return
-		}
-	case v1alpha1.KindStressChaos:
-		if err := s.createStressChaos(exp); err != nil {
-			c.Status(http.StatusInternalServerError)
-			_ = c.Error(utils.ErrInternalServer.WrapWithNoMessage(err))
-			return
-		}
-	default:
+	createFuncs := map[string]actionFunc{
+		v1alpha1.KindPodChaos:     s.createPodChaos,
+		v1alpha1.KindNetworkChaos: s.createNetworkChaos,
+		v1alpha1.KindIOChaos:      s.createIOChaos,
+		v1alpha1.KindStressChaos:  s.createStressChaos,
+		v1alpha1.KindTimeChaos:    s.createTimeChaos,
+		v1alpha1.KindKernelChaos:  s.createKernelChaos,
+	}
+
+	f, ok := createFuncs[exp.Target.Kind]
+	if !ok {
 		c.Status(http.StatusBadRequest)
 		_ = c.Error(utils.ErrInvalidRequest.New("Target kind is not available"))
```

Comment:
How about changing to "_ = c.Error(utils.ErrInvalidRequest.New(exp.Target.Kind + " is not supported")) "?